Sources/SwiftToolchain/ToolchainManagement.swift

```diff
@@ -291,4 +284,27 @@ extension FileSystem {
 
     return result.sorted()
   }
+  
+  public func fetchLocalSwiftVersion() throws -> String? {
+    guard let cwd = currentWorkingDirectory else { return nil }
+
+    let versionFile = cwd.appending(component: ".swift-version")
```

Comment:
Please generalize this as `var versionFilePath: AbsolutePath` computed property, similarly to how it was done for `swiftenvVersionsPath` and `cartonSDKPath` above. It would be fair to `fatalError` if `currentWorkingDirectory` returns `nil`, this will allow making the return type of `versionFilePath` non-optional.